tests/testthat/_snaps/verb-pivot-wider.md

```diff
@@ -0,0 +1,55 @@
+# can pivot all cols to wide
+
+    Code
+      lazy_frame(key = c("x", "y", "z"), val = 1:3) %>% dbplyr_pivot_wider_spec(spec)
+    Output
+      <SQL>
+      SELECT MAX(CASE WHEN (`key` = 'x') THEN (`val`) END) AS `x`, MAX(CASE WHEN (`key` = 'y') THEN (`val`) END) AS `y`, MAX(CASE WHEN (`key` = 'z') THEN (`val`) END) AS `z`
+      FROM `df`
+
+# implicit missings turn into explicit missings
+
+    Code
+      lazy_frame(a = 1:2, key = c("x", "y"), val = 1:2) %>% dbplyr_pivot_wider_spec(
+        spec)
+    Output
+      <SQL>
+      SELECT `a`, MAX(CASE WHEN (`key` = 'x') THEN (`val`) END) AS `x`, MAX(CASE WHEN (`key` = 'y') THEN (`val`) END) AS `y`
+      FROM `df`
+      GROUP BY `a`
+
+# values_fn can be a single function
+
+    Code
+      dbplyr_pivot_wider_spec(df, spec1, values_fn = sum)
+    Output
+      <SQL>
+      SELECT `a`, SUM(CASE WHEN (`key` = 'x') THEN (`val`) END) AS `x`
+      FROM `df`
+      GROUP BY `a`
+
+# values_fn cannot be NULL
+
+    Code
+      expect_error(dbplyr_pivot_wider_spec(df, spec1, values_fn = NULL))
+
+# can fill in missing cells
+
+    Code
+      dbplyr_pivot_wider_spec(df_lazy, spec, values_fill = 0)
+    Output
+      <SQL>
+      SELECT `g`, `name`, `value`, MAX(CASE WHEN (`key` = 'x') THEN (`val`) WHEN NOT(`key` = 'x') THEN (0.0) END) AS `x`, MAX(CASE WHEN (`key` = 'y') THEN (`val`) WHEN NOT(`key` = 'y') THEN (0.0) END) AS `y`
```

Comment:
I wonder if this should be `COALESCE(MAX(CASE WHEN ...), 0.0)`? What should happen if `fill = 1` and `values_fn = sum` and there are multiple missing values?

Reply:
I am not sure I understand correctly what you mean but I think it is explicitly wanted to not use `coalesce()`. At least I adapted this test from `tidyr`

```r
test_that("values_fill only affects missing cells", {
  df <- memdb_frame(g = c(1, 2), name = c("x", "y"), value = c(1, NA))
  out <- tidyr::pivot_wider(df, values_fill = 0) %>%
    collect()
  expect_equal(out$y, c(0, NA))
})
```

Or here directly with `tidyr`:

``` r
library(tidyr)
df <- tibble(g = c(1, 2, 2), name = c("x", "y", "y"), value = c(1, NA, 2))
df
#> # A tibble: 3 x 3
#>       g name  value
#>   <dbl> <chr> <dbl>
#> 1     1 x         1
#> 2     2 y        NA
#> 3     2 y         2

pivot_wider(df, values_fill = 1, values_fn = sum)
#> # A tibble: 2 x 3
#>       g     x     y
#>   <dbl> <dbl> <dbl>
#> 1     1     1     1
#> 2     2     1    NA

pivot_wider(df, values_fill = 1, values_fn = function(x) sum(x, na.rm = TRUE))
#> # A tibble: 2 x 3
#>       g     x     y
#>   <dbl> <dbl> <dbl>
#> 1     1     1     1
#> 2     2     1     2
```

So, `values_fill` actually ignores `NA`. Does this answer your question or did I misunderstand you?

Reply:
Ok, cool, that puts my mind at rest.